packages/programs/src/programs/DefaultProgram.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { IBaseWorld, WorldConsumer } from "@latticexyz/world-consumer/src/experimental/WorldConsumer.sol";
+
+import { EntityId } from "@dust/world/src/EntityId.sol";
+
+import { ReversePlayer } from "@dust/world/src/codegen/tables/ReversePlayer.sol";
+
+import { IAttachProgramHook, IDetachProgramHook } from "@dust/world/src/ProgramInterfaces.sol";
+
+import { Admin } from "../codegen/tables/Admin.sol";
+import { AllowedPlayers } from "../codegen/tables/AllowedPlayers.sol";
+
+/**
+ * @title DefaultProgram
+ */
+contract DefaultProgram is IAttachProgramHook, IDetachProgramHook, WorldConsumer {
+  /**
+   * @notice Initializes the DefaultProgram
+   * @param _world The world contract
+   */
+  constructor(IBaseWorld _world) WorldConsumer(_world) { }
+
+  function onAttachProgram(EntityId caller, EntityId target, bytes memory extraData) external onlyWorld {
+    address player = ReversePlayer.get(caller);
+    require(player != address(0), "Caller is not a player");
```

Comment:
hmm I wonder if we should just use owner entityIds directly, as maybe some other entity might want to use these programs?

Reply:
made it use entity ids